From 87b7370e578598059862336a91d5792dc82bd3bf Mon Sep 17 00:00:00 2001
From: PixelyIon <pixelyion@protonmail.com>
Date: Sun, 21 Nov 2021 15:06:09 +0530
Subject: [PATCH 06/14] Implement Skyline Proxy Settings

Settings values are to be handled by proxy as well by Skyline code, it was determined to do things in this manner it was rather pointless to have the settings struct which has been now eliminated and been replaced with a namespace of the same name.
---
 common/settings.h | 33 +++++++++++++++++----------------
 1 file changed, 17 insertions(+), 16 deletions(-)

diff --git a/common/settings.h b/common/settings.h
index 3d3c225..c1f9ca7 100644
--- a/common/settings.h
+++ b/common/settings.h
@@ -5,21 +5,22 @@
 
 #include "common_types.h"
 
-namespace Shader {
-    struct Settings {
-        struct Values {
-            bool renderer_debug{};
-            bool disable_shader_loop_safety_checks{};
-            struct ResolutionScalingInfo {
-                u32 up_scale{1};
-                u32 down_shift{0};
-                f32 up_factor{1.0f};
-                f32 down_factor{1.0f};
-                bool active{};
-                bool downscale{};
-            } resolution_info{};
-        };
-
-        static Values values;
+namespace Shader::Settings {
+    /**
+     * @note Only contains the settings relevant to the shader compiler
+     */
+    struct Values {
+        bool renderer_debug;
+        bool disable_shader_loop_safety_checks;
+        struct ResolutionScalingInfo {
+            u32 up_scale{1};
+            u32 down_shift{0};
+            f32 up_factor{1.0f};
+            f32 down_factor{1.0f};
+            bool active{};
+            bool downscale{};
+        } resolution_info;
     };
+
+    static inline Values values{}; //!< A static structure with the values set by Skyline code
 }
-- 
2.38.1

